### PR TITLE
docs: update usage doc with info about `--watch` flag and absence of `--version` flag

### PR DESCRIPTION
**Pull request by eelcofolkertsma**
To reflect changes in asyncapi generate fromTemplate command for --watch option

Documentation update to follow `asyncapi generate fromTemplate --help`

### CODE REVIEW

**Review by derberg — APPROVED** [review]
lgtm, thanks!

just updated the PR title and removed `fix` as this is docs update, so no need to trigger patch release

**Comment by derberg** [comment]
@Florence-Njeri hey, please have a look and approve

**Comment by Florence-Njeri** [comment]
@derberg taking a look

File: docs/usage.md [hunk 1]
```diff
@@ -8,24 +8,29 @@ There are two ways to use the generator:
...
+Generates whatever you want using templates compatible with AsyncAPI Generator.
 ```bash
-Usage: asyncapi generate fromTemplate <asyncapi> <template> [<options>]
+USAGE
+  $ asyncapi generate fromTemplate [ASYNCAPI] [TEMPLATE] [-h] [-d <value>] [-i] [--debug] [-n <value>] [-o <value>] [--force-write] [-w] [-p <value>] [--map-base-url <value>]
 
-- <asyncapi>: Local path or URL pointing to AsyncAPI document for example https://bit.ly/asyncapi
-- <template>: Name of the generator template like for example @asyncapi/html-template or https://github.com/asyncapi/html-template
+ARGUMENTS
+  ASYNCAPI  - Local path, url or context-name pointing to AsyncAPI file
```
Comment (Florence-Njeri) [inline docs/usage.md:17]:
```suggestion
  ASYNCAPI  - Local path, URL pointing to AsyncAPI document  for example https://bit.ly/asyncapi
```

File: docs/usage.md [hunk 2]
```diff
@@ -149,4 +154,4 @@ try {
 }
 ```
 
-See the [API documentation](api) for more examples and full API reference information.
+See the [API documentation](/docs/tools/generator/api) for more examples and full API reference information.
```
Comment (Florence-Njeri) [inline docs/usage.md:157]: This leads to a 404 error. We should leave it as is @derberg any thought about this?

File: docs/usage.md [hunk 3]
```diff
@@ -8,24 +8,29 @@ There are two ways to use the generator:
...
 ```bash
-Usage: asyncapi generate fromTemplate <asyncapi> <template> [<options>]
+USAGE
+  $ asyncapi generate fromTemplate [ASYNCAPI] [TEMPLATE] [-h] [-d <value>] [-i] [--debug] [-n <value>] [-o <value>] [--force-write] [-w] [-p <value>] [--map-base-url <value>]
 
-- <asyncapi>: Local path or URL pointing to AsyncAPI document for example https://bit.ly/asyncapi
-- <template>: Name of the generator template like for example @asyncapi/html-template or https://github.com/asyncapi/html-template
+ARGUMENTS
+  ASYNCAPI  - Local path, url or context-name pointing to AsyncAPI file
+  TEMPLATE  - Name of the generator template like for example @asyncapi/html-template or https://github.com/asyncapi/html-template
```
Comment (Florence-Njeri) [inline docs/usage.md:19]: @derberg correct me if I am wrong but I feel the changes being made to this file might be unnecessary especially this part `asyncapi generate fromTemplate [ASYNCAPI] [TEMPLATE] [-h] [-d <value>] [-i] [--debug] [-n <value>] [-o <value>] [--force-write] [-w] [-p <value>] [--map-base-url <value>]` why not just leave it as was `asyncapi generate fromTemplate <asyncapi> <template> [<options>]`

File: docs/usage.md (hunk 2 above)
Comment (Florence-Njeri) [inline docs/usage.md:157]:
```suggestion
See the [API documentation](api) for more examples and full API reference information.
```

File: docs/usage.md (hunk 1 above)
Comment (eelcofolkertsma) [reply docs/usage.md:17]: Hi @Florence-Njeri , @derberg 
I did "construct" my update by copy/pasting output of `asyncapi generate fromTemplate --help`. The suggestions break alignment between code and documentation. Do you really want to go that direction?

Alternative: apply suggestions to code (help text) and documentation in parallel. @derberg, can you point me to the help text?

File: docs/usage.md (hunk 2 above)
Comment (eelcofolkertsma) [reply docs/usage.md]: I do not know how/why link was incorrect, but I have accepted suggestion so now reference is working again

File: docs/usage.md (hunk 3 above)
Comment (eelcofolkertsma) [reply docs/usage.md:19]: See earlier argument. Keep documentation in sync with help-output generated from code

**Comment by eelcofolkertsma** [comment]
This is my second PR ever, so apologies where I behave in unexpected ways. Please coach me

**Comment by Florence-Njeri** [comment]
Hey @eelcofolkertsma no, you are doing amazing and the continuous improvements to our docs are a great help to our community.

File: docs/usage.md (hunk 2 above)
Comment (derberg) [reply docs/usage.md]: @Florence-Njeri actually both are good - not sure where you see 404

I personally prefer full relative links like `/docs/tools/generator/api` but `api` works as well. Let's leave it as `api` as my approach for full relative links is an old habit that I follow but brainfart every time I try to find in my brain the reason for doing it 😄

File: docs/usage.md (hunk 1 above)
Comment (derberg) [reply docs/usage.md:17]: @Florence-Njeri normally I like that you want to clarify help with examples, like example link. Problem is that https://bit.ly/asyncapi is no longer the best as it points to asyncapi `2.0.0` and nobody knows who owns this short link and how to update it. So we should not promote it

so correct help is
`Local path, name of CLI context or URL pointing to AsyncAPI document`

Comment (derberg) [reply docs/usage.md:17]: @eelcofolkertsma we can accept the update here now and if you could followup with PR to fix in in CLI in https://github.com/asyncapi/cli/blob/master/src/commands/generate/fromTemplate.ts#L106 that would be awesome

File: docs/usage.md (hunk 3 above)
Comment (derberg) [reply docs/usage.md:19]: yeah, I think in this particular case good to follow the output from CLI that is following best practices

**Comment by derberg** [comment]
@eelcofolkertsma great contributions so far 👏🏼

**Review by Florence-Njeri — APPROVED** [review]
Hey @eelcofolkertsma I took another look at your PR and it looks good. Good job and thank you for providing some extra context in the comments

**Pull request merged**